parmed/openmm/parameters.py

```diff
@@ -257,11 +264,18 @@ def _write_omm_atom_types(self, dest, skip_types):
 
     def _write_omm_residues(self, dest, skip_residues):
         if not self.residues: return
+        written_residues = set()
         dest.write(' <Residues>\n')
         for name, residue in iteritems(self.residues):
-            if not isinstance(residue, ResidueTemplate) or name in skip_residues:
+            if (not isinstance(residue, ResidueTemplate) or name in skip_residues
+                or self._templhasher(residue) in written_residues):
```

Comment:
`self._templhasher` might be expensive -- especially when generalized to larger residue templates.  Can you precompute this just inside the `for` loop to avoid calling this function many times?